### PR TITLE
Add option for encryption iteration/octet count

### PR DESCRIPTION
**Pull request by geraintluff**
I wanted to be able to alter the iteration count when encrypting the private-key, so I added an option for it.

Existing tests still pass - I wasn't sure where to put a new test or what it should be, though.

### CODE REVIEW

File: src/type/s2k.js [hunk 1]
```diff
@@ -57,6 +57,23 @@ S2K.prototype.get_count = function () {
...
+  while (upperBits < 15 && count > (31 << (upperBits + expbias))) {
+    upperBits++;
+  }
+  
+  var lowerBits = 0;
+  while (lowerBits < 15 && count > (16 + lowerBits) << (upperBits + expbias)) {
+    lowerBits++;
+  }
+
+  this.c = lowerBits + (upperBits << 4);
```
Comment (tanx) [inline src/type/s2k.js]: I honestly cannot judge with the naked eye if this is correct or not. Perhaps add some unit test with several test vectors and please document your reasoning behind the code.

File: src/type/s2k.js [hunk 2]
```diff
@@ -57,6 +57,23 @@ S2K.prototype.get_count = function () {
...
+  return this.get_count();
```
Comment (tanx) [inline src/type/s2k.js]: No need to return the getter in a setter function.

File: src/packet/secret_key.js [hunk 3]
```diff
@@ -175,7 +175,7 @@ SecretKey.prototype.write = function () {
  * This can be used to remove passphrase protection after calling decrypt().
  * @param {String} passphrase
  */
-SecretKey.prototype.encrypt = function (passphrase) {
+SecretKey.prototype.encrypt = function (passphrase, options) {
```
Comment (tanx) [inline src/packet/secret_key.js]: just using a second `count` parameter here is fine. No need to use an options object.

**Comment by tanx** [comment]
Thanks for the PR. I made some comments. Please rebase to the current master and push your branch against to initiate a new travis build. Thanks.

**Comment by bartbutler** [comment]
Closing due to inactivity.

**Pull request closed without merging**